launchpod/angular-launchpod/backend/src/main/java/com/google/launchpod/servlets/FileUploadServlet.java

```diff
@@ -146,29 +122,29 @@ public void doPost(HttpServletRequest req, HttpServletResponse res) throws Illeg
       desiredFeedEntity = datastore.get(entityKey);
     } catch (IllegalArgumentException e) {
       // If entityId cannot be converted into a key
-      writeResponse(res, "Sorry, this is not a valid id.", HttpServletResponse.SC_BAD_REQUEST);
+      Helper.writeResponse(res, "Sorry, this is not a valid id.", HttpServletResponse.SC_BAD_REQUEST);
       return;
     } catch (EntityNotFoundException e) {
       // No matching entity in Datastore
-      writeResponse(res, "Your entity could not be found.", HttpServletResponse.SC_NOT_FOUND);
+      Helper.writeResponse(res, "Your entity could not be found.", HttpServletResponse.SC_NOT_FOUND);
       return;
     }
     String mp3Link = makeMp3Link(id);
 
     // Create embedded entity to store MP3 data in desired entity as a property
     EmbeddedEntity mp3 = new EmbeddedEntity();
-    mp3.setProperty(ID, id);
-    mp3.setProperty(MP3_LINK, mp3Link);
-    mp3.setProperty(EMAIL, email);
-    desiredFeedEntity.setProperty(MP3, mp3);
+    mp3.setProperty(Keys.ID, id);
+    mp3.setProperty(Keys.MP3_LINK, mp3Link);
+    mp3.setProperty(Keys.USER_EMAIL, email);
+    desiredFeedEntity.setProperty(Keys.MP3, mp3);
 
-    String xmlString = (String) desiredFeedEntity.getProperty(XML_STRING);
+    String xmlString = (String) desiredFeedEntity.getProperty(Keys.XML_STRING);
 
     // Modify the xml string
     RSS rssFeed = XML_MAPPER.readValue(xmlString, RSS.class);
     Channel channel = rssFeed.getChannel();
       
-    String entityEmail = (String) desiredFeedEntity.getProperty(EMAIL);
+    String entityEmail = (String) desiredFeedEntity.getProperty(Keys.USER_EMAIL);
 
     // Verify that user is modifying a feed they created
     if (entityEmail.equals(email)) {
```

Comment:
nit: instead of creating a channel variable , you could just do rssFeed.getChannel().addItem(new Item())

Reply:
makes sense!